tests/cephadm/test_remove_cluster.py

```diff
@@ -0,0 +1,98 @@
+    if result:
+        log.info("Disabled cephadm module sucessfully")
+        return
+
+    raise CephadmOpsFailure("Failed to disable cephadm module")
+
+
+def _get_cluster_fsid(installer):
+    """Get cluster FSID"""
+    cmd = "cephadm shell -- ceph fsid"
+    result = installer.exec_command(sudo=True, cmd=cmd)
+
+    if result:
+        log.info("Cluster FSID: {}".format(result[0]))
+        return result[0].strip()
+
+    raise CephadmOpsFailure("Failed to get cephadm cluster FSID")
+
+
+def _remove_cephadm_cluster(nodes, fsid):
+    """Remove cephadm cluster from nodes"""
+    cmd = "cephadm rm-cluster --fsid {} --zap-osds --force".format(fsid)
+
+    for node in nodes:
+        result = node.exec_command(sudo=True, cmd=cmd)
+        if result:
+            log.info("Removed cluster from node {}".format(str(node)))
+        else:
+            raise CephadmOpsFailure("Failed to remove cephadm cluster")
+
+
+def _validate_cluster_node_cleanup(nodes):
+    """Validate cleaned cluster"""
+    var_lib_ceph = "/var/lib/ceph"
+    var_lib_cephadm = "/var/lib/cephadm"
+    ceph_dir = "find {} -maxdepth 0 -empty -exec echo {{}} is empty. \\;"
+    podman_ps = "podman ps | grep rhceph || echo 'RHCEPH containers not present'"
+    lsblk_ceph = "lsblk | grep ceph || echo 'Ceph disks are not present'"
+
+    for node in nodes:
+        result = node.exec_command(sudo=True, cmd=ceph_dir.format(var_lib_ceph))
+        result = node.exec_command(sudo=True, cmd=ceph_dir.format(var_lib_cephadm))
+        result = node.exec_command(sudo=True, cmd=podman_ps)
+        log.info("podman status - '{}'".format(result))
+        if not result[0].strip() == "RHCEPH containers not present":
+            raise CephadmOpsFailure("Containers are not cleaned")
+
+        result = node.exec_command(sudo=True, cmd=lsblk_ceph)
+        log.info("ceph disk status - '{}'".format(result))
+        if not result[0].strip() == "Ceph disks are not present":
+            raise CephadmOpsFailure("Disks are not cleaned")
+
+
+def _validate_cluster_installer_cleanup(installer):
+    etc_ceph = "/etc/ceph"
+    ceph_dir = "find {} -maxdepth 0 -empty -exec echo {{}} is empty. \\;"
+
+    ceph_status = "cephadm shell -- ceph -s || echo 'No ceph cluster present'"
+
+    result = installer.exec_command(sudo=True, cmd=ceph_dir.format(etc_ceph))
+    log.info("'{}' dir status '{}'".format(etc_ceph, result))
+
+    result = installer.exec_command(sudo=True, cmd=ceph_status)
+    log.info("Ceph status - '{}'".format(result))
+    if not result[0].strip() == "No ceph cluster present":
+        raise CephadmOpsFailure("Failed to clean ceph cluster")
+
+
+def run(ceph_cluster, **kw):
+    """Cephadm rm-cluster"""
+    nodes = ceph_cluster.get_nodes()
+    nodes = nodes if isinstance(nodes, list) else [nodes]
+
+    installer = ceph_cluster.get_ceph_object("installer")
+
+    fsid = _get_cluster_fsid(installer)
+
+    _disable_mgr_module(installer)
```

Comment:
Can you avoid _ in the beginning of every method?

Reply:
@sunilangadi2 these are internal methods to module, not inted to be used outside. Python Pep8 style guide indicate to use `_` for the same - https://peps.python.org/pep-0008/#descriptive-naming-styles